EIPS/eip-4844.md

```diff
@@ -0,0 +1,558 @@
+Instead of embedding the full contents in the body, the contents of the blobs are propagated separately, as a "sidecar".
+
+This "sidecar" design provides forward compatibility for further data increases:
+with data-availability-sampling (DAS) not all blobs will have to be downloaded by all beacon nodes on the network.
+
+### BeaconBlockBody extension
+
+We add to the `BeaconBlockBody` a new object, `blob_kzgs: List[KZGCommitment, MAX_OBJECT_LIST_SIZE]`,
+and update `BeaconBlock` and `SignedBeaconBlock` to reflect this updated `BeaconBlockBody`.
+
+At the end of the `process_block` state-transition, the `BeaconBlock` and its contained `ExecutionPayload` 
+need to be cross-validated with `cross_validate(block.body)`.
+
+```python
+def cross_validate(body: BeaconBlockBody):
+    all_versioned_hashes = []
+    for tx in body.execution_payload.blob_transactions:
+        all_versioned_hashes.extend(tx.message.blob_versioned_hashes)
+    assert all_versioned_hashes == [
+        kzg_to_versioned_hash(kzg) for kzg in body.blob_kzgs
+    ]
+```
+
+### BlobsSidecar
+
+We introduce a `BlobsSidecar` container:
+
+```python
+class BlobsSidecar(Container):
+    beacon_block_root: Root
+    beacon_block_slot: Slot
+    blobs: List[Blob, MAX_OBJECT_LIST_SIZE]
+```
+
+Nodes broadcast the `SignedBeaconBlock` along with its corresponding
+`BlobsSidecar` over the network instead of plain beacon blocks.
+
+### Required blobs downloading
+
+When a node receives both objects, it first calls `verify_blobs` and if this call passes it runs the usual processing on the beacon block.
+
+```python
+def verify_blobs(signed_block: SignedBeaconBlock, blobs_sidecar: BlobsSidecar):
+    assert signed_block.message.slot == blobs_sidecar.beacon_block_slot
+    assert hash_tree_root(signed_block.message) == blobs_sidecar.beacon_block_root
+    kzgs = signed_block.message.body.blob_kzgs
+    blobs = blobs_sidecar.blobs
+    assert len(kzgs) == len(blobs)
+    for kzg, blob in zip(kzgs, blobs):
+        assert blob_to_kzg(blob) == kzg
+```
+
+A verified `BlobsSidecar` is a strict requirement for the validity of the beacon block.
+
+Without the sidecar the block may be processed but MUST NOT be considered valid until a valid `BlobsSidecar` has been downloaded.
+This download requirement can be substituted with an availability-check in the future with data-availability-sampling.
+
+### P2P changes
+
+Full peer-to-peer specification is out of scope for this EIP, the consensus-specs expands on this instead.
+
+In summary, the peer to peer layer changes involve:
+- Along other consensus-layer upgrade modifications to `SignedBeaconBlock`, the `beacon_block` gossip topic is updated 
+  and the `beacon_blocks_by_range/2/` and `beacon_blocks_by_root/2/` type tables are updated.
+- Blobs sidecars are relayed through a new global GossipSub topic: `blobs_sidecar`
+- Blobs sidecars are synced through ReqResp protocols: 
```

Comment:
Important to define how long these are expected to be served (e.g. on the order of a month?) to put the bound on when this new data can be pruned